src/mdns/mdns_avahi.hpp

```diff
@@ -335,22 +350,38 @@ class PublisherAvahi : public Publisher
 
     typedef std::vector<Service> Services;
 
+    struct Host
+    {
+        std::string      mHostName;
+        AvahiAddress     mAddress = {};
+        AvahiEntryGroup *mGroup   = nullptr;
+        Services         mServices;
+        bool             mIsCommitted = false;
+    };
+
+    typedef std::vector<Host> Hosts;
+
     static void HandleClientState(AvahiClient *aClient, AvahiClientState aState, void *aContext);
     void        HandleClientState(AvahiClient *aClient, AvahiClientState aState);
 
-    void        CreateGroup(AvahiClient *aClient);
+    otbrError GetOrCreateHost(AvahiClient *aClient, const char *aHostName, Host **aHost);
+    void      ResetHost(Host *aHost);
+
+    otbrError   CreateGroup(AvahiClient *aClient, AvahiEntryGroup **aGroup);
     static void HandleGroupState(AvahiEntryGroup *aGroup, AvahiEntryGroupState aState, void *aContext);
     void        HandleGroupState(AvahiEntryGroup *aGroup, AvahiEntryGroupState aState);
 
-    Services         mServices;
-    AvahiClient *    mClient;
-    AvahiEntryGroup *mGroup;
-    Poller           mPoller;
-    int              mProtocol;
-    const char *     mDomain;
-    State            mState;
-    StateHandler     mStateHandler;
-    void *           mContext;
+    otbrError MakeFullName(char *aFullName, size_t aFullNameLength, const char *aName);
+
+    AvahiClient *mClient;
+    Host         mDefaultHost;
+    Hosts        mHosts;
```

Comment:
I think we just need a `mServicesGroup`.

Reply:
Could elaborate more? Do you just propose to change the name (why?) or a different structure of the data?

Reply:
I mean we can keep the `AvahiEntryGroup *mGroup;`, but just rename it to `AvahiEntryGroup *mServicesGroup;`

Reply:
I'll check if it will remain valid after the rework we talked above, but I changed it to `Host` as we not only needed the avahi group, but also other host-related fields like services connected with the host. But probably I'll organize it differently if we go with "group-per-service" approach :)

Reply:
@Damian-Nordic Thanks, Looking forward to the change.